lib/oli_web/controllers/collaborator_controller.ex

```diff
@@ -26,45 +27,88 @@ defmodule OliWeb.CollaboratorController do
           |> Enum.map(&String.trim/1)
           |> Enum.filter(fn e -> e != "" end)
 
-        if Enum.count(emails) > @max_invitation_emails do
-          conn
-          |> put_flash(
-            :error,
-            "Collaborator invitations cannot exceed #{@max_invitation_emails} emails at a time. Please try again with fewer invites"
-          )
-          |> redirect(to: Routes.live_path(OliWeb.Endpoint, OliWeb.Projects.OverviewLive, project_id))
-        else
-          emails
-          |> Enum.reduce({conn, []}, fn email, {conn, failures} ->
-            add_collaborator(conn, email, project_id, failures)
+        authors =
+          authors_emails
+          |> String.split(",")
+          |> Enum.map(&String.trim/1)
+          |> Enum.filter(fn e -> e != "" end)
+
+        is_collaborator? =
+          Enum.all?(emails, fn elem ->
+            Enum.member?(authors, elem)
```

Comment:
Since we are not giving detailed feedback about the email/emails that is/are already a collaborator, we could stop the iteration as soon as we get the first email that is contained in the authors list. So, a `reduce_while` could avoid the full iteration in many cases (except the case where the last email element is the one contained in the author's list)

`Enum.reduce_while(emails, false, fn email, acc -> if email not in authors, do: {:cont, acc}, else: {:halt, true} end)`

Reply:
I understand what you are telling me and it seems like a great solution, but in this part of the code I want to identify if all the emails entered to send invitations are included in the existing collaborators. 

In the code, below, there is a case in which some of the emails match with some existing collaborator then another error message is displayed.